GitHub workflow failing in activate script

**Description**

Error reported in branch [build #1220](https://github.com/thelastpickle/cassandra-medusa/actions/runs/800385038) during `Run tox step` in building with py 3.6, 3.7, and 3.8.

**Error**
```text
/home/runner/work/_temp/d2b360bc-945a-4ede-8913-342560b4bd47.sh: line 5: unexpected argument `]]' to conditional unary operator
Error: Process completed with exit code 2.
```

**Inspection**

Fails on an invalid condition check in the `venv/bin/activate` script due to `coveralls` support introduced in #337.

The `venv/bin/activate` invocation shows the following during build time.

```bash
  # Report partial code coverage
  if [[ -n  ]];
  then
    coveralls
  fi
```

Initial inspection shows that the secrets token for coveralls is not available, which creates an invalid condition check.

```bash
 if [[ -n ${{ secrets.COVERALLS_REPO_TOKEN }} ]];
 then
   coveralls
 fi
```

--- a/.coveragerc
+++ b/.coveragerc
@@ -1,0 +1,13 @@
+[run]
+branch = True
+source = medusa
+
+[report]
+exclude_lines =
+    if self.debug:
+    pragma: no cover
+    raise NotImplementedError
+    if __name__ == .__main__.:
+ignore_errors = True
+omit =
+    tests/*

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -36,7 +36,7 @@ jobs:
         python -m pip install --upgrade pip
         pip install -r requirements.txt
         pip install tox
-        pip install coveralls
+
     - name: Lint with flake8
       run: |
         pip install flake8
@@ -46,17 +46,13 @@ jobs:
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --ignore=W503
     - name: Run tox
       env:
-        COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
         . venv/bin/activate
         tox -e ${{ matrix.tox-py }}
 
-        # Report partial code coverage
-        if [[ -n ${{ secrets.COVERALLS_REPO_TOKEN }} ]];
-        then
-          coveralls
-        fi
+    - uses: codecov/codecov-action@v1
+      name: Report code coverage  
 
   debian-build:
     # Build debian packages
@@ -177,12 +173,11 @@ jobs:
         pip install -r requirements.txt
         pip install -r requirements-test.txt
         pip install ccm
-        pip install coveralls
-    - name: Check if integration tests and coveralls can run
+
+    - name: Check if integration tests can run
       env:
         AWS_ACCESS_KEY_ID: ${{ secrets.BUCKET_KEY }}
         AWS_SECRET_ACCESS_KEY: ${{ secrets.BUCKET_SECRET }}
-        COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
       run: |
         if [[ ( -n "${AWS_ACCESS_KEY_ID}" && "${{ matrix.it-backend }}" == "s3" ) \
           || ( "${{ matrix.it-backend }}" == "local" ) \
@@ -196,20 +191,11 @@ jobs:
           echo "IT_CAN_RUN=no" >> $GITHUB_ENV
         fi 
 
-        # Check if code coverage can be reported
-        if [[ -n ${{ env.COVERALLS_REPO_TOKEN }} ]];
-        then
-          echo "COVERALLS_CAN_RUN=yes" >> $GITHUB_ENV
-        else
-          echo "COVERALLS_CAN_RUN=no" >> $GITHUB_ENV
-        fi
-
     - name: Run integration tests
       if: ${{ env.IT_CAN_RUN == 'yes' }}
       env:
         AWS_ACCESS_KEY_ID: ${{ secrets.BUCKET_KEY }}
         AWS_SECRET_ACCESS_KEY: ${{ secrets.BUCKET_SECRET }}
-        COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
       run: |
         set -e
         . venv/bin/activate
@@ -268,23 +254,13 @@ jobs:
           ./run_integration_tests.sh -v --cassandra-version=${{ matrix.cassandra-version }}
         fi
 
-        # Report partial code coverage
-        if [[ "${{ env.COVERALLS_CAN_RUN }}" == "yes" ]];
-        then
-          mv tests/integration/.coverage .
-          coveralls
-        fi
+        # Move and convert the coverage analysis file to XML
+        mv tests/integration/.coverage .
+        coverage xml
+
+    - uses: codecov/codecov-action@v1
+      name: Report code coverage
   
-  finish-coveralls:
-    needs: integration-tests
-    runs-on: ubuntu-latest
-    steps:
-    - name: Coveralls Finished
-      if: ${{ env.COVERALLS_CAN_RUN }} == 'yes'
-      uses: coverallsapp/github-action@master
-      with:
-        github-token: ${{ secrets.GITHUB_TOKEN }}
-        parallel-finished: true
     
   publish-debian:
     # We can only release if the build above succeeded first

--- a/.gitignore
+++ b/.gitignore
@@ -27,3 +27,4 @@ installed-by-dh_installdocs
 debian/.debhelper
 debian/cassandra-medusa/
 .coverage
+coverage.xml

--- a/README.md
+++ b/README.md
@@ -19,7 +19,7 @@
 
 [![Hosted By: Cloudsmith](https://img.shields.io/badge/OSS%20hosting%20by-cloudsmith-blue?logo=cloudsmith&style=flat-square)](https://cloudsmith.io/~thelastpickle/repos/medusa/packages/)
 
-[![Coverage Status](https://coveralls.io/repos/github/thelastpickle/cassandra-medusa/badge.svg?branch=master)](https://coveralls.io/github/thelastpickle/cassandra-medusa?branch=master)
+[![codecov](https://codecov.io/gh/thelastpickle/cassandra-medusa/branch/master/graph/badge.svg?token=KTDCRD82NU)](https://codecov.io/gh/thelastpickle/cassandra-medusa)
 
 Medusa for Apache Cassandra&trade;
 ==================================

--- a/tox.ini
+++ b/tox.ini
@@ -12,7 +12,7 @@ deps =
 commands =
     python setup.py check -m -s
     flake8 . --ignore=W503,E402
-    pytest --cov=medusa -v tests/
+    pytest --cov=medusa --cov-report=xml -v tests/
 
 [flake8]
 exclude = .tox,*.egg,build,data,scripts,env,venv

